transports/dns/src/lib.rs

```diff
@@ -404,30 +357,33 @@ pub enum DnsErr<TErr> {
     TooManyLookups,
 }
 
-impl<TErr> fmt::Display for DnsErr<TErr>
+#[deprecated(note = "Use `Error` instead.")]
+pub type DnsError<TErr> = Error<TErr>;
```

Comment:
I am so glad that we have `cargo semver-checks` which spotted this really subtle mistake that the original type was called `DnsErr`! :)

https://github.com/libp2p/rust-libp2p/actions/runs/6256289255/job/16989149563#step:4:95

cc @obi1kenobi